src/MatrixClient.ts

```diff
@@ -371,11 +371,11 @@ export class MatrixClient extends EventEmitter {
     /**
      * Sets the presence status for the current user.
      * @param {"online"|"offline"|"unavailable"} presence The new presence state for the user.
-     * @param {string} statusMessage Optional status message to include with the presence.
+     * @param {string?} statusMessage Optional status message to include with the presence.
      * @returns {Promise<any>} Resolves when complete.
      */
     @timedMatrixClientFunctionCall()
-    public async setPresenceStatus(presence: "online" | "offline" | "unavailable", statusMessage: string = null): Promise<any> {
+    public async setPresenceStatus(presence: "online" | "offline" | "unavailable", statusMessage: string | undefined): Promise<any> {
```

Comment:
to maintain backwards compatibility:
```suggestion
    public async setPresenceStatus(presence: "online" | "offline" | "unavailable", statusMessage: string = undefined): Promise<any> {
```

Reply:
to maintain backwards compatibility and correct typing:
```suggestion
    public async setPresenceStatus(presence: "online" | "offline" | "unavailable", statusMessage: string | undefined = undefined): Promise<any> {
```

Reply:
technically this project doesn't have correct typing on lots of things (yet) :p